lib/src/layer/tile_layer/tile_painter.dart

```diff
@@ -0,0 +1,45 @@
+import 'dart:ui' as ui;
+import 'package:flutter/rendering.dart';
+import 'package:flutter_map/src/layer/tile_layer/tile_model.dart';
+
+///Draws the tile images on the canvas.
+class TilePainter extends CustomPainter {
+  /// The list of tile models.
+  List<TileModel> tiles;
+
+  /// Constructs a `TilePainter` with the given list of `TileModel` objects.
+  TilePainter({required this.tiles});
+
+  @override
+  void paint(Canvas canvas, Size size) {
+    for (final tile in tiles) {
+      //Simplify tile positions and sizes
+      final double left = tile.tileImage.coordinates.x * tile.scaledTileSize -
+          tile.currentPixelOrigin.x;
+      final double top = tile.tileImage.coordinates.y * tile.scaledTileSize -
+          tile.currentPixelOrigin.y;
+      final double width = tile.scaledTileSize;
+      final double height = tile.scaledTileSize;
```

Comment:
Can be moved to the if scope to avoid useless compute if imageInfo is null.

Reply:
Thanks! Well spotted. I adjusted for that.